src/ol/format/Feature.js

```diff
@@ -113,7 +113,7 @@ import {
  * {@link module:ol/Feature~Feature} objects from a variety of commonly used geospatial
  * file formats.  See the documentation for each format for more details.
  *
- * @template {import('../Feature.js').FeatureClass} [T=typeof import('../Feature.js').default]
+ * @template {import('../Feature.js').FeatureClass} [FeatureClassType=import('./Feature.js').FeatureToFeatureClass<import("../Feature.js").default>]
```

Comment:
Maybe you already tried this and it was a dead end, but it seems like it would be simpler if the template here were just this.

```
/**
 * @template {import('../Feature.js').default} T
 */
```

Or maybe it needs to be optional, not sure.

Then below, we could have this:
```js
    /**
     * @protected
     * @type {import('../Feature.js').FeatureClass}
     */
    this.featureClass = Feature;
```

I'm wondering if this would keep us from having to use `import('./Feature.js').FeatureToFeatureClass<import("../Feature.js").default>` in so many other places.

Reply:
If we did that, we'd lose the ability to infer the return type of `readFeatures()` etc. to infer the feature type from the configured `featureClass`.